cassandra-protocol/src/types.rs

```diff
@@ -43,6 +48,181 @@ pub mod prelude {
     pub use crate::types::AsRustType;
 }
 
+pub trait AsCassandraType {
+    fn as_cassandra_type(&self) -> CDRSResult<Option<CassandraType>>;
+
+    fn get_wrapper_fn(&self, col_type: ColTypeOption) -> Box<dyn Fn(&CBytes) -> CassandraType> {
```

Comment:
This fn doesn't use self, so it would be better to make it a free fn or an associated fn. In fact, wouldn't it be better to return a pointer to an already existing free fn for each option, given those lambdas do not capture anything and are the same always? This will avoid dynamic allocations.

Reply:
Moved the wrapper functions to a separate module and got rid of the unnecessary allocations.